Automattic/wp-calypso
# Update <MyPlanCard /> to make it more versatile and easier to re-use

This PR updates the `<MyPlanCard />` component - props in particular - to make it more versatile and easier to re-use on the `/plans/my-plan` page.

**Before**
![Screenshot 2019-11-14 at 18 00 39](https://user-images.githubusercontent.com/478735/68954169-be3e6a00-07c3-11ea-87f2-aac4ce8db50c.png)

**After**
![Screenshot 2019-11-15 at 16 19 10](https://user-images.githubusercontent.com/478735/68954166-bbdc1000-07c3-11ea-88ed-3fbc0dd75c2d.png)

#### Changes proposed in this Pull Request

* Use a single `action` prop instead of the `buttonLabel` and `buttonTarget` pair
* Use `details` instead of `expirationDate` to make it more versatile (e.g. display renew information)
* Accept an `isError` flag so that the details element turns red (match production)
* Accept an `isPlaceholder` flag for when the expiration date and/or action is loaded to show pulsating grey rectangles instead

#### Testing instructions



* ~Check out this branch~
* ~Go to http://calypso.localhost:3000/devdocs/design/my-plan-card~
* ~Confirm there is no regression in how the component looks and function~
* ~Switch to see placeholders/loading state and confirm the component looks good~

**Update: Since this component has been [moved away from the `components` folder](https://github.com/Automattic/wp-calypso/pull/37637/commits/4ac916c9f63b590dfba76647f23ae63d66362b42), there's no way to test it out in the devdocs. Instead, it will be made available for previewing and testing in the [PR where it's being integrated with /plans/my-plan` page](https://github.com/Automattic/wp-calypso/pull/37642).**

Fixes n/a
